From 486ab95c53a0694c6218e00f690ad1ff470359a5 Mon Sep 17 00:00:00 2001
From: Lukas Holecek <hluk@email.cz>
Date: Fri, 29 Oct 2021 15:11:57 +0200
Subject: [PATCH 2/2] Fix Windows build

---
 src/kstatusnotifieritem.cpp | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/src/kstatusnotifieritem.cpp b/src/kstatusnotifieritem.cpp
index e4e19db..38eefc8 100644
--- a/src/kstatusnotifieritem.cpp
+++ b/src/kstatusnotifieritem.cpp
@@ -655,7 +655,11 @@ void KStatusNotifierItem::activate(const QPoint &pos)
 
 QString KStatusNotifierItem::providedToken() const
 {
+#ifdef QT_DBUS_LIB
     return d->statusNotifierItemDBus->m_xdgActivationToken;
+#else
+    return {};
+#endif
 }
 
 bool KStatusNotifierItemPrivate::checkVisibility(QPoint pos, bool perform)
-- 
2.31.1

